File: ts/project/.sdk/src/cmp/clojure/fragment/EntityCreateOp.fragment.clj
; EJECT-START
;; Create a new EntityName (reqdata: body data).
(defn create [ent reqdata ctrl]
  (let [ctx (core/make-context
             (vs/jm "opname" "create" "ctrl" ctrl
                    "match" (deref (:_match ent)) "data" (deref (:_data ent)) "reqdata" reqdata)
             (:_entctx ent))]
    (run-op ctx
            (fn []
              (when-let [result (core/oget ctx :result)]
                (when (core/oget result :resdata)
                  (reset! (:_data ent)
                          (let [m (core/to-map (vs/clone (core/oget result :resdata)))] (if m m (vs/jm))))))))))
; EJECT-END

File: ts/project/.sdk/src/cmp/elixir/fragment/EntityCreateOp.fragment.ex
# EJECT-START

  # Returns the created entityname entity map (ProjectName.Types.entityname/0)
  # on success; pipeline errors surface as the error value built by
  # Utility.make_error (shape is utility-configurable), hence term().
  @spec create(map(), ProjectName.Types.entityname_create_data() | nil, map() | nil) :: term()
  def create(ent, reqdata, ctrl \\ nil) do
    ctx =
      Context.new(
        S.jm([
          "opname", "create",
          "ctrl", ctrl,
          "match", S.getprop(ent, "_match"),
          "data", S.getprop(ent, "_data"),
          "reqdata", reqdata
        ]),
        S.getprop(ent, "_entctx")
      )

    post_done = fn ->
      result = S.getprop(ctx, "result")

      if result != nil do
        rd = S.getprop(result, "resdata")
        if rd != nil, do: S.setprop(ent, "_data", H.or_(H.to_map(S.clone(rd)), S.jm([])))
      end
    end

    Pipeline.run_op(ctx, post_done)
  end

# EJECT-END
